testdata/templates/example_project_create_empty_project_id.json

```diff
@@ -0,0 +1,34 @@
+[
+  {
+    "name": "//cloudbilling.googleapis.com/projects/{{.Provider.project}}/billingInfo",
+    "asset_type": "cloudbilling.googleapis.com/ProjectBillingInfo",
+    "ancestry_path": "{{.Ancestry}}/project/{{.Provider.project}}",
+    "resource": {
+      "version": "v1",
+      "discovery_document_uri": "https://www.googleapis.com/discovery/v1/apis/cloudbilling/v1/rest",
+      "discovery_name": "ProjectBillingInfo",
+      "parent": "//cloudresourcemanager.googleapis.com/projects/{{.Provider.project}}",
+      "data": {
+        "billingAccountName": "billingAccounts/{{.Project.BillingAccountName}}",
+        "name": "projects//billingInfo"
```

Comment:
If the placeholder logic for project id goes into assetName - I think you could use assetName to build the billing account name, which I think would fix this bug.

Reply:
asset name and project ID seem to two separate logic.  assetName issue I addressed in file transport.go, and if getting project attempts all failed, I returned a placeholder;  while this billing data it directly used d.Get("project_id") in project.go.

Reply:
yes, the implementation currently uses `d.Get("project_id")` - but I think that's wrong, since it would also fail if the project_id was supposed to default to the provider project. I think replacing [this line](https://github.com/GoogleCloudPlatform/terraform-validator/blob/f72ddec9686bcc79a994171c9fce556efa9e497e/converters/google/resources/project.go#L139) with `name = assetName("projects/{{project_id_or_project}}/billingInfo")` would handle the fallbacks correctly, including generating a placeholder. (It would just be a little confusing since it's not actually making an asset name.)

I would have a slight preference for making this change if it's simple and just works, but otherwise we could do that work in a separate ticket.

Reply:
changed to split the asset name by `/`, and get the project name, and pass the project name to billing object build function

Reply:
that seems like a good solution. 👍